tokio-net/src/driver/registration.rs

```diff
@@ -39,64 +37,20 @@ use std::{io, ptr, usize};
 /// [`register`]: #method.register
 /// [`poll_read_ready`]: #method.poll_read_ready`]
 /// [`poll_write_ready`]: #method.poll_write_ready`]
-#[derive(Debug)]
+#[derive(Debug, Default)]
 pub struct Registration {
-    /// Stores the handle. Once set, the value is not changed.
-    ///
-    /// Setting this requires acquiring the lock from state.
-    inner: UnsafeCell<Option<Inner>>,
-
-    /// Tracks the state of the registration.
-    ///
-    /// The least significant 2 bits are used to track the lifecycle of the
-    /// registration. The rest of the `state` variable is a pointer to tasks
-    /// that must be notified once the lock is released.
-    state: AtomicUsize,
+    inner: Option<Inner>,
```

Comment:
It should be possible to avoid the `Option<Inner>` indirection by performing the `mio` registration in `Registration::new`.